src/extension.ts

```diff
@@ -199,6 +202,69 @@ export async function activate(context: vscode.ExtensionContext, perfStats: { lo
     ext.managedIdentityBranchDataProvider = new ManagedIdentityBranchDataProvider();
     ext.v2.api.resources.registerAzureResourceBranchDataProvider(AzExtResourceType.ManagedIdentityUserAssignedIdentities, ext.managedIdentityBranchDataProvider);
 
+
+    const AUTH_PROVIDER_ID = 'microsoft'; // VS Code Azure auth provider
+    const SCOPES = ['https://management.azure.com/.default']; // Default ARM scope
+
+    registerEvent(
+        'treeView.onDidChangeSessions',
+        vscode.authentication.onDidChangeSessions,
+        async (context: IActionContext, _e: vscode.AuthenticationSessionsChangeEvent) => {
+            context.errorHandling.suppressDisplay = true;
+            context.telemetry.suppressIfSuccessful = true;
+            context.telemetry.properties.isActivationEvent = 'true';
+
+            try {
+                const accounts = await vscode.authentication.getAccounts(AUTH_PROVIDER_ID);
+                if (accounts.length === 0) {
+                    ext.outputChannel.appendLine('No Azure accounts found.');
+                    return;
+                }
+
+                const session = await vscode.authentication.getSession(
+                    AUTH_PROVIDER_ID,
+                    SCOPES);
+
+                if (!session) {
+                    ext.outputChannel.appendLine('No session available for Azure account.');
+                    return;
+                }
+
+                // Default tenant from session
+                const defaultTenantId = session.account.id.split('.')[1];
+
+                // Check if extension context args contain tenant override
+                const tenantFromArg = vscode.workspace.getConfiguration().get<string>('@azure.argTenant');
+
+                const effectiveTenantId = tenantFromArg || defaultTenantId;
+
+                // AuthenticationRecord structure
+                const authRecord = {
+                    username: session.account.label,
+                    authority: 'https://login.microsoftonline.com', // VS Code auth provider default
+                    homeAccountId: `${session.account.id}`,
+                    tenantId: effectiveTenantId,
+                    clientId: "aebc6443-996d-45c2-90f0-388ff96faa56" // VS Code client ID
+                };
+
+
+                // Write to a well-known location in .azure home directory
+                const authDir = path.join(os.homedir(), '.azure', 'vscode-auth', 'vscode-azureresourcegroups');
+                const authRecordPath = path.join(authDir, 'authRecord.json');
+
+                // Ensure directory exists
+                fs.mkdirSync(authDir, { recursive: true });
```

Comment:
Better to do this and `writeFile` async. You can use `fs-extra` package for async versions of all the things `fs` provides.

Reply:
added